src/vs/workbench/services/textfile/common/textFileSaveParticipant.ts

```diff
@@ -32,25 +36,41 @@ export class TextFileSaveParticipant extends Disposable {
 		// undoStop before participation
 		model.textEditorModel?.pushStackElement();
 
-		for (const saveParticipant of this.saveParticipants) {
-			if (token.isCancellationRequested || !model.textEditorModel /* disposed */) {
-				break;
-			}
+		const cts = new CancellationTokenSource(token);
```

Comment:
Should we still be using the `progress` parameter to report that we are in the "running save participants" phase here?

Reply:
https://github.com/microsoft/vscode/pull/230291#discussion_r1784289747